src/core_plugins/choropleth/public/choropleth_controller.js

```diff
@@ -0,0 +1,156 @@
+module.controller('KbnChoroplethController', function ($scope, $element, Private, Notifier, getAppState,
+                                                       serviceSettings, config) {
+
+  const tooltipFormatter = Private(AggResponsePointSeriesTooltipFormatterProvider);
+  const ResizeChecker = Private(ResizeCheckerProvider);
+  const notify = new Notifier({ location: 'Vectormap' });
+
+  const resizeChecker = new ResizeChecker($element);
+  const containerNode = $element[0];
+
+  let kibanaMap = null;
+  resizeChecker.on('resize', () => {
+    if (kibanaMap) {
+      kibanaMap.resize();
+    }
+  });
+  let choroplethLayer = null;
+
+  const kibanaMapReady = makeKibanaMap();
+  $scope.$watch('esResponse', async function (response) {
+    kibanaMapReady.then(() => {
+      const metricsAgg = _.first($scope.vis.aggs.bySchemaName.metric);
+      const termAggId = _.first(_.pluck($scope.vis.aggs.bySchemaName.segment, 'id'));
+
+      let results;
+      if (!response || !response.aggregations) {
+        results = [];
+      } else {
+        const buckets = response.aggregations[termAggId].buckets;
+        results = buckets.map((bucket) => {
+          return {
+            term: bucket.key,
+            value: metricsAgg.getValue(bucket)
+          };
+        });
+      }
+
+      if (!$scope.vis.params.selectedJoinField) {
+        $scope.vis.params.selectedJoinField = $scope.vis.params.selectedLayer.fields[0];
+      }
+      updateChoroplethLayer($scope.vis.params.selectedLayer.url);
+      choroplethLayer.setMetrics(results, metricsAgg);
+      setTooltipFormatter();
+
+      kibanaMap.useUiStateFromVisualization($scope.vis);
+      kibanaMap.resize();
+      $element.trigger('renderComplete');
+    });
+  });
+
+  $scope.$watch('vis.params', (visParams) => {
+    kibanaMapReady.then(() => {
+      if (!visParams.selectedJoinField) {
+        visParams.selectedJoinField = visParams.selectedLayer.fields[0];
+      }
+
+      updateChoroplethLayer(visParams.selectedLayer.url);
+      choroplethLayer.setJoinField(visParams.selectedJoinField.name);
+      choroplethLayer.setColorRamp(truncatedColorMaps[visParams.colorSchema]);
+      setTooltipFormatter();
+
+      kibanaMap.setShowTooltip(visParams.addTooltip);
+      kibanaMap.setLegendPosition(visParams.legendPosition);
+      kibanaMap.useUiStateFromVisualization($scope.vis);
+      kibanaMap.resize();
+      $element.trigger('renderComplete');
+    });
+  });
+
+  getAvailableVectorLayers()
+    .then(function (layersFromService) {
+      //todo: disable play-button until these settings are loaded!
+      $scope.vis.type.params.vectorLayers = $scope.vis.type.params.vectorLayers.concat(layersFromService);
+      $scope.$apply();
+    });
+
+  async function makeKibanaMap() {
+
+    const tmsSettings = await serviceSettings.getTMSService();
+    const minMaxZoom = tmsSettings.getMinMaxZoom(false);
+    kibanaMap = new KibanaMap(containerNode, minMaxZoom);
+    kibanaMap.setZoomLevel(2);
```

Comment:
it's a default, that works well on most screens. but that shouldn't be configured here, so I moved it to the kibana_map.